src/dune_pkg/rev_store.ml

```diff
@@ -63,9 +63,11 @@ let with_flock lock_path ~f =
               Unix_error.Detailed.create ue ~syscall:"flock" ~arg:"unlock"
               |> Unix_error.Detailed.raise)
       | Ok `Failure ->
-        Code_error.raise
-          (sprintf "Couldn't acquire lock after %d attempts to lock" max_retries)
-          []
+        let pid = Io.read_file lock_path |> Int.of_string_exn in
+        User_error.raise
+          ~hints:[ Pp.textf "Check if process %d is running and close it" pid ]
```

Comment:
This isn't the hint I would give usually. It could be that another process is running and rightfully so, in which case it doesn't make sense to close it. We should already have a message like this for dune.lock:
```
        User_error.raise
          [ Pp.textf
              "A running dune%s instance has locked the build directory. If this is not \
               the case, please delete %s"
              (match Io.read_file (Path.build lock_file) with
               | exception _ -> ""
               | pid -> sprintf " (pid: %s)" pid)
              (Path.Build.to_string_maybe_quoted lock_file)
          ])
;;
```